src/execution/index.ts

```diff
@@ -219,112 +184,44 @@ async function sendTransactionAsUserOp(
       targetChain: targetChain.id,
     } as TransactionResult
   }
-
-  const accountAccessList = sourceChain
-    ? {
-        chainIds: [sourceChain.id as SupportedChain],
-      }
-    : getDefaultAccountAccessList()
-
-  const metaIntent: MetaIntent = {
-    targetChainId: targetChain.id,
-    tokenTransfers: tokenRequests.map((tokenRequest) => ({
-      tokenAddress: tokenRequest.address,
-      amount: tokenRequest.amount,
-    })),
-    targetAccount: accountAddress,
-    targetGasUnits: gasLimit,
-    userOp: getEmptyUserOp(),
-    accountAccessList,
-  }
-
-  const orchestrator = getOrchestratorByChain(
-    targetChain.id,
+  const orderPath = await getUserOpOrderPath(
+    sourceChain,
+    targetChain,
+    tokenRequests,
+    accountAddress,
+    gasLimit,
```

Comment:
Did a major refactor here by reusing the low-level functions as much as possible